capreolus/searcher/__init__.py

```diff
@@ -50,7 +51,8 @@ def _anserini_query_from_file(self, topicsfn, anserini_param_str, output_base_pa
             logger.debug(f"skipping Anserini SearchCollection call because path already exists: {donefn}")
             return
 
-        self["index"].create_index()
+        if not self["index"].exists():
```

Comment:
The task is now responsible for making sure that the searcher index is built. Ideally I'd have liked this to happen in the constructor, but as discussed it would end up being annoying in notebooks